spec/dummy/db/schema.rb

```diff
@@ -119,6 +119,23 @@
    FROM demography.citizens;
   SQL
 
+  add_foreign_key "demography.cities", "demography.countries", :name => "demography_cities_country_id_fk", :column => "country_id", :exclude_index => true
+
+  add_foreign_key "demography.citizens", "public.users", :name => "demography_citizens_user_id_fk", :column => "user_id", :exclude_index => true
+
+  add_foreign_key "pets", "public.users", :name => "pets_user_id_fk", :column => "user_id", :exclude_index => true
+
+  create_function 'public.pets_not_empty()', :boolean, <<-DEFINITION
+BEGIN
+  IF (SELECT COUNT(*) FROM pets) > 0
+  THEN
+    RETURN true;
+  ELSE
+    RETURN false;
+  END IF;
+END;
+DEFINITION
```

Comment:
See handy trick above to make code read better with indentation while not actually introducing leading whitespace.